nginx/nginx.conf

```diff
@@ -0,0 +1,84 @@
+# vim: syntax=nginx ts=4 sw=4 et
+include /etc/nginx/modules-enabled/*;
+
+worker_processes  1;
+error_log  /dev/stdout warn;
+env ENV;
+
+events {
+    worker_connections  4096;
+}
+
+http {
+    include       mime.types;
+    default_type  text/html;
+
+    # requied for shared_lua_dict
+    map_hash_bucket_size 128;
+
+    # hide the version in the Server header
+    server_tokens off;
+
+    # use real client ip rather than ALB address
+    set_real_ip_from 10.0.0.0/8;
+    set_real_ip_from 172.16.0.0/12;
+    set_real_ip_from 192.168.0.0/16;
+    real_ip_header X-Forwarded-For;
+
+    # apache-style combined log format
+    log_format  main  '$remote_addr - $remote_user [$time_local] "$request" '
+                      '$status $body_bytes_sent "$http_referer" '
+                      '"$http_user_agent"';
+
+    access_log  /dev/stdout  main;
+
+    sendfile        on;
+    tcp_nopush      on;
+    tcp_nodelay     on;
+    keepalive_timeout  130;
+    gzip  on;
+
+    # microcaching
+    # The proxy_cache_path will be used to store responses for 1s on select
+    # endpoints. This improves performance with minimal impact.
+    proxy_cache_path /tmp/cache keys_zone=cache:10m levels=1:2 inactive=600s max_size=100m;
+
+    lua_shared_dict upstream 24k;
+    init_by_lua_block {
+        local env = os.getenv("ENV");
+        ngx.shared.upstream:set("env", env)
+        ngx.log(ngx.WARN, "Starting WNYC 3000 Vue web client Nginx with ENV=" .. env)
+    }
+
+    upstream nuxt {
+        server 127.0.0.1:3000 max_fails=0;
+    }
+
+    server {
+        listen       80;
+        server_name  _;
+        resolver     172.16.0.2 valid=60s;
+
+        # This is the docker resolver, it can be used to debug upstreams.
+        # resolver     127.0.0.11 valid=10s;
+
+        location / {
+            include headers.conf;
+            include microcache.conf;
+            proxy_pass http://nuxt;
+            rewrite ^/(.*)/$ /$1 permanent;
+        }
+
+       # test endpoint
+        location /_test {
+            include headers.conf;
+            add_header Content-Type application/json;
+            return 200 '{"status": "ok"}';
+        }
+
+        location /_health {
```

Comment:
this endpoint would ideally be implemented in the Vue app and just return a 200. It makes the healthcheck more robust in that it checks to see that the Vue app is running whereas in its current state it just checks that nginx is running.

Reply:
we just need to add a route for _health? let me know, I'll do this

Reply:
In the old fastboot world, the healthcheck was written into the nypr-fastboot library. I'm not sure if we'll have something comparable with vue, but the default healthcheck route, the sentry installation, and a metrics collection agent were all installed by a dependency on that parent repo. I will eventually want these things to exist with Nuxt and Vue, but it doesn't need to block the deployment to a staging env, so for now, this half-ass healthcheck is fine.